[19.0][MIG] website_attribute_set

bosd opened the pull request:
depends on #232

kobros-tech committed:
[ADD] website_attribute_set: add a module to show custom attributes in website

kobros-tech committed:
[IMP] website_attribute_set: show attributes in product page in e-commerce app

kobros-tech committed:
[IMP] website_attribute_set: show all attributes in filter bar

kobros-tech committed:
[IMP] website_attribute_set: remember selected attribute values in filter bar

kobros-tech committed:
[IMP] website_attribute_set: enable filter for additional attributes

kobros-tech committed:
[IMP] website_attribute_set: enable search for additional attributes

kobros-tech committed:
[IMP] website_attribute_set: display additional attributes in products comparison

kobros-tech committed:
[IMP] website_attribute_set: display reference numbers in products comparison

kobros-tech committed:
[FIX] website_attribute_set: some improvements

kobros-tech committed:
[IMP] website_attribute_set: show attributes in a nice badge

kobros-tech committed:
[IMP] website_attribute_set: show attributes in specification section

kobros-tech committed:
[IMP] website_attribute_set: format attributes values in comparison

kobros-tech committed:
[IMP] website_attribute_set: set a default field val

kobros-tech committed:
[IMP] website_attribute_set: add test cases

Commit:
[IMP] website_attribute_set: pre-commit auto fixes

MiquelRForgeFlow commented on website_attribute_set/controllers/main.py:
```diff
@@ -0,0 +1,421 @@
+
+class WebsiteSale(main.WebsiteSale):
+    @route()
+    def shop(
```
It's better to do a `super` call than to do a total method replace.

Commit:
[MIG] website_attribute_set: migrate to 19.0

Key changes for Odoo 19 compatibility:
- Update version to 19.0.1.0.0
- Fix pricelist access: use request.env.user.property_product_pricelist
  instead of website.pricelist_id
- Update _shop_lookup_products signature (removed attrib_set parameter)
- Replace SQL-based price aggregation with search_read approach
- Fix TableCompute usage: use main.TableCompute().process() directly
- Remove unnecessary lazy wrappers from products_prices and get_product_prices
- Rename _get_additional_extra_shop_values to _get_additional_shop_values
- Remove visible_on_ecommerce filter from product tags (field removed)
- Update domain validation to work with Odoo 19 ir.model.fields constraint
- Fix test assertions to expect ValidationError instead of ValueError

Commit:
[IMP] website_attribute_set: add HTTP controller tests

Add integration tests for website controllers to detect server errors
early. These tests verify that the shop page and product pages can be
accessed without 500 errors, which helps catch issues like missing
attributes (e.g., website.pricelist_id) during development.

Commit:
[ADD] test-requirements.txt: add CI test dependencies

Add test requirements file to reference the pending PRs for
attribute_set and product_attribute_set dependencies that are
not yet merged into the 19.0 branch.

Commit:
[IMP] website_attribute_set: fix attribute filter and add demo data

- Fix filter logic to properly apply additional attribute filters
  - Add _get_shop_domain override using Odoo 19 Domain.AND API
  - Fix _search_get_details to handle different attribute types
    (boolean, select, integer, float, char/text)
- Add JavaScript to trigger form submission on filter change
- Add demo data with sample products and attributes for testing
- Add filter tests for boolean and select attribute types

Commit:
[FIX] website_attribute_set: use unique demo attribute names

Rename demo attributes to use x_was_demo_* prefix to avoid
conflicts with existing attributes in the database.

Commit:
[FIX] website_attribute_set: fix search with filter conflict

Fix TypeError when using search with additional attribute filters.
The issue was passing **post to _get_shop_domain which included
'search' key, conflicting with the positional 'search' argument.

Add test to catch this error early.

Commit:
[IMP] website_attribute_set: add e_com_searchable field

Add a new boolean field 'e_com_searchable' to control which attributes
are included in e-commerce search, separate from visibility.

This allows large text fields (like descriptions) to be displayed on
product pages without impacting search performance.

- Add e_com_searchable field to attribute.attribute model
- Update search_extra() to filter by e_com_searchable instead of e_com_visibility
- Add field to attribute form view (only visible when e_com_visibility is True)
- Update demo data with searchable flags (description field is NOT searchable)
- Update and add tests for the new behavior

Commit:
[IMP] website_attribute_set: add caching for attribute lookups

Add ormcache to _get_ecom_visible_attribute_ids to avoid repeated
database queries when displaying multiple products with the same
attribute set on the shop page.

- Cache attribute IDs by model_name, attribute_set_id, and include_native flag
- Clear cache on attribute create/write/unlink when visibility-related fields change
- Significantly reduces database queries on shop pages with many products

Commit:
[IMP] website_attribute_set: add advanced filter options

Add per-attribute configuration for advanced filter features:

1. Range filters (e_com_range_filter):
   - For integer/float attributes, show min/max inputs instead of dropdown
   - Filters products within the specified range

2. Multi-value selection (e_com_multi_select):
   - For select/multiselect attributes, show checkboxes instead of dropdown
   - Allows selecting multiple values with OR logic

3. Product count badges (e_com_show_count):
   - Shows the number of matching products next to each filter option

Features:
- All options configurable per-attribute in the attribute form view
- Updated filter templates with conditional rendering
- Updated controller with refactored domain building for better maintainability
- Updated JavaScript to handle range input changes
- Demo data includes examples of all features

JordiBForgeFlow commented:
Superseded by https://github.com/OCA/odoo-pim/pull/250
